apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -39,1347 +38,1419 @@ const pgErrors = {
+    }));
+
+    const fixedInserts = rawInserts.map((insert, idx) => {
+      const result = { ...insert};
+      const columns = columnsList[idx];
+      result.data = result.data.map((obj) => {
+        return _.mapValues(obj, (value, key) => {
+          const column = columns.find((c) => c.columnName === key);
+          // fix: we used to serialize arrays before this, now we pass them as
+          // json arrays properly
+          return this.normalizeValue(value, column.dataType);
+        })
+      })
+      return result;
+    })
 
-          pid = null;
+    await this.driverExecuteSingle(buildInsertQueries(this.knex, fixedInserts).join(";"));
 
-          if(!data) {
-            return []
-          }
+    return true;
+  }
 
-          return data
-        } catch (err) {
-          if (canceling && err.code === pgErrors.CANCELED) {
-            canceling = false;
-            err.sqlectronError = 'CANCELED_BY_USER';
-          }
+  private async updateValues(rawUpdates: TableUpdate[]): Promise<TableUpdateResult[]> {
+    const updates = rawUpdates.map((update) => {
+      const result = { ...update };
+      result.value = this.normalizeValue(update.value, update.columnType);
+      return result;
+    })
+    log.info("applying updates", updates);
+    let results: TableUpdateResult[] = [];
+    await this.driverExecuteSingle(buildUpdateQueries(this.knex, updates).join(";"));
+    // NOTE (@day): this could be a weird issue, we shall see
+    const data = await this.driverExecuteSingle(buildSelectQueriesFromUpdates(this.knex, updates).join(";"));
+    results = [data.rows[0]];
+
+    return results;
+  }
 
-          throw err;
-        } finally {
-          cancelable.discard();
-        }
-      });
-    },
+  private async deleteRows(deletes: TableDelete[]) {
+    await this.driverExecuteSingle(buildDeleteQueries(this.knex, deletes).join(";"));
 
-    async cancel(): Promise<void> {
-      if (!pid) {
-        throw new Error('Query not ready to be canceled');
+    return true
+  }
+
+  /**
+   * Gets the version details for the connection.
+   *
+   * Example version strings:
+   * CockroachDB CCL v1.1.0 (linux amd64, built 2017/10/12 14:50:18, go1.8.3)
+   * CockroachDB CCL v20.1.1 (x86_64-unknown-linux-gnu, built 2020/05/19 14:46:06, go1.13.9)
+   *
+   * PostgreSQL 9.4.26 on x86_64-pc-linux-gnu (Debian 9.4.26-1.pgdg90+1), compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit
+   * PostgreSQL 12.3 (Debian 12.3-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
+   *
+   * PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.12103
+   */
+  private async getVersion(): Promise<VersionInfo> {
+    const { version } = (await this.driverExecuteSingle("select version()")).rows[0]
+
+    if (!version) {
+      return {
+        version: '',
+        isPostgres: false,
+        isCockroach: false,
+        isRedshift: false,
```

Comment:
I think we probably don't need these since we have different classes for each database.

Reply:
Yeah I'm a moron, completely forgot about these